HW0/P4/P4.py

```diff
@@ -40,38 +46,125 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    data2 = np.loadtxt('P4_measurements.txt', delimiter=",")
+    C = np.matrix([[1.0/rx, 0, 0], [0, 1.0/ry, 0], [0, 0, 1.0/rz]])
+    #print C
+    result = np.dot(data2, C)
+    x_coords = np.asarray(result[:, 0]).reshape(-1)
+    y_coords = np.asarray(result[:, 1]).reshape(-1)
+    z_coords = np.asarray(result[:, 2]).reshape(-1)
+    # print z_coords
+
+
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt, 0,0],[0,0,0,0, 1-c*dt,0],[0,0,0,0,0,1-c*dt]])
+    a = np.matrix([0,0,0,0,0,g*dt]).T
+    s = np.matrix([0,0,2,15,3.5,4.0]).T
+
+    preds = np.zeros([6, K])
+
+    for i in xrange(K):
+        s_i = np.dot(A, s) + a
+        for j in xrange(len(s_i)):
+            preds[j, i] = s_i[j]
+        s = s_i
+
+
+    x_coords = preds[0]
+    y_coords = preds[1]
+    z_coords = preds[2]
 
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+            '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.matrix([[bx,0,0,0,0,0], [0,by,0,0,0,0], [0,0,bz,0,0,0], [0,0,0,bvx,0,0], [0,0,0,0,bvy,0], [0,0,0,0,0,bvz]])
+    C = np.matrix([[rx, 0, 0,0,0,0], [0, ry, 0,0,0,0], [0, 0, rz,0,0,0]])
+
+    measures = np.loadtxt('P4_measurements.txt', delimiter=",")
+
+    def predictS(A, s_k, a):
+        # print A.shape
+        # print s_k.shape
+        # print a.shape
+        return np.dot(A, s_k) + a
+
+    def predictSig(Sig_k, A, B):
+        return np.linalg.inv(np.dot(np.dot(A, Sig_k), A.T) + np.dot(B, B.T))
+
+    def updateSig(Sig, C):
+        return np.linalg.inv(Sig + np.dot(C.T, C))
+
+    def updateS(Sig_k1, Sig, s, C, m):
+        a = (np.dot(Sig, s) + np.dot(C.T, m))
+        b = Sig_k1
+        return np.dot(b, a)
```

Comment:
for these four functions, you could consider not passing in the A, B... as parameters. This will make the style a little more brief. Nothing big but this would help if you have a lot of parameters for a function.